### PR TITLE
Simplify API for plugins 

### PR DESCRIPTION
**Pull request by Creatone**
Instead of task_measurements/task_labels/task_resources - should have 
```task: {measurements: {}, labels: {}}``` or other better option.

### CODE REVIEW

File: wca/runners/allocation.py [hunk 1]
```diff
@@ -284,17 +284,16 @@ def _initialize_rdt(self) -> bool:
...
         """Allocator callback body."""
 
         current_allocations = _get_tasks_allocations(containers)
 
         # Allocator callback
         allocate_start = time.time()
         new_allocations, anomalies, extra_metrics = self._allocator.allocate(
-            platform, tasks_measurements, tasks_resources, tasks_labels,
-            current_allocations)
+            platform, tasks_data, current_allocations)
```
Comment (mmucek95) [inline wca/runners/allocation.py]: Allocate method takes only platform and task_data arguments.

File: wca/runners/measurement.py [hunk 2]
```diff
@@ -414,28 +410,35 @@ def _prepare_tasks_data(containers: Dict[Task, Container]) -> \
...
+        tasks_data[task.task_id] = {
+                TaskDataType.LABELS: {},
+                TaskDataType.MEASUREMENTS: {},
+                TaskDataType.RESOURCES: {}
+                }
 
-    return tasks_measurements, tasks_resources, tasks_labels
+        tasks_data[task.task_id][TaskDataType.LABELS] = task_labels
+        tasks_data[task.task_id][TaskDataType.MEASUREMENTS] = task_measurements
+        tasks_data[task.task_id][TaskDataType.RESOURCES] = task_resources
```
Comment (mmucek95) [inline wca/runners/measurement.py]: I would suggest simplifying the code to somethig like this:
tasks_data[task.task_id] = {
TaskDataType.LABELS: task_labels,
TaskDataType.MEASUREMENTS: task_measurements,
TaskDataType.RESOURCES: task_resources
}

File: wca/runners/measurement.py [hunk 3]
```diff
@@ -315,16 +315,15 @@ def _iterate(self):
 
         try:
             # Tasks data
```
Comment (felidadae) [inline wca/runners/measurement.py]: we do not need that comment

File: wca/runners/measurement.py [hunk 4]
```diff
@@ -413,30 +409,35 @@ def _prepare_tasks_data(containers: Dict[Task, Container]) -> \
         if TaskResource.MEM in task.resources:
             task_measurements[MetricName.MEM.value] = task.resources[TaskResource.MEM.value]
 
-        task_labels = task.labels.copy()
+        tasks_data[task.task_id] = TaskData(
```
Comment (felidadae) [inline wca/runners/measurement.py]: not better to create constructor which takes as input Task? just to discuss

File: wca/detectors.py [hunk 5]
```diff
@@ -34,10 +34,19 @@
...
-TasksResources = Dict[TaskId, Dict[str, float]]
-TaskLabels = Dict[str, str]
-TasksLabels = Dict[TaskId, TaskLabels]
+
+
+TaskMeasurements = Measurements
+TaskAllocations = Dict[str, str]
+
+
+@dataclass
```
Comment (felidadae) [inline wca/detectors.py]: should find better place than wca/detectors.py as that structure is bothed used in detector and allocator.

File: wca/runners/measurement.py (hunk 4 above)
Comment (felidadae) [reply wca/runners/measurement.py]: would be more implicit that TaskData is derived from class Task

File: wca/detectors.py (hunk 5 above)
Comment (felidadae) [reply wca/detectors.py]: however i dont know if there is better place to put that

File: wca/runners/allocation.py [hunk 6]
```diff
@@ -374,19 +373,19 @@ def _iterate_body(self,
         allocations_package.send(common_labels)
 
 
-def _get_tasks_allocations(containers) -> TasksAllocations:
-    tasks_allocations: TasksAllocations = {}
+def _get_tasks_allocations(containers, tasks_data: TasksData) -> TasksAllocations:
```
Comment (felidadae) [inline wca/runners/allocation.py]: Just add comment that input parameter tasks_data is modified inside function

File: wca/runners/measurement.py (hunk 4 above)
Comment (Creatone) [reply wca/runners/measurement.py]: Discussed offline.

File: tests/testing.py [hunk 7]
```diff
@@ -318,7 +334,9 @@ def _decorated_function(*args, **kwargs):
                    return_value=(platform_mock, [metric('platform-cpu-usage')], {})), \
              patch('wca.platforms.collect_topology_information', return_value=(1, 1, 1)), \
              patch('wca.security.are_privileges_sufficient', return_value=True), \
-             patch('resource.getrusage', return_value=Mock(ru_maxrss=WCA_MEMORY_USAGE)):
+             patch('resource.getrusage', return_value=Mock(ru_maxrss=WCA_MEMORY_USAGE)), \
```
Comment (felidadae) [inline tests/testing.py]: Add comment in commit that you are solving bug by the way

File: tests/testing.py [hunk 8]
```diff
@@ -318,7 +334,9 @@ def _decorated_function(*args, **kwargs):
...
+             patch('wca.perf_uncore.UncorePerfCounters._open'), \
+             patch('wca.perf.PerfCounters._open'):
```
Comment (felidadae) [inline tests/testing.py]: Add comment in commit that you are solving bug by the way.

**Pull request merged**